front/components/members/InvitationModal.tsx

```diff
@@ -258,28 +259,43 @@ export function InviteEmailModal({
 export function EditInvitationModal({
   owner,
   invitation,
-  isOpen,
-  setOpen,
+  onClose,
 }: {
   owner: WorkspaceType;
   invitation: MembershipInvitationType;
-  isOpen: boolean;
-  setOpen: (open: boolean) => void;
+  onClose: () => void;
 }) {
+  const [selectedRole, setSelectedRole] = useState<ActiveRoleType>(
+    invitation.initialRole
+  );
   const sendNotification = useContext(SendNotificationsContext);
   const confirm = useContext(ConfirmContext);
 
   return (
-    <Modal
-      title="Edit Invitation"
-      isOpen={isOpen}
+    <ElementModal
+      title="Edit invitation"
+      openOnElement={invitation}
       onClose={() => {
-        setOpen(false);
+        onClose();
+        setSelectedRole(invitation.initialRole);
       }}
-      saveLabel="Save"
-      savingLabel="Saving..."
-      hasChanged={false}
+      hasChanged={selectedRole !== invitation.initialRole}
       variant="side-sm"
+      onSave={async (closeModalFn) => {
+        if (!selectedRole) {
```

Comment:
This feels weird, what does this case represent?

Reply:
Leftover, ✂️ it